beacon_node/lighthouse_network/src/rpc/codec/ssz_snappy.rs

```diff
@@ -1102,8 +1025,7 @@ mod tests {
         // A MetaDataV2 still encodes as a MetaDataV1 since version is Version::V1
         assert_eq!(
             encode_then_decode_response(
-                Protocol::MetaData,
-                Version::V1,
+                SupportedProtocol::MetaDataV1,
                 RPCCodedResponse::Success(RPCResponse::MetaData(metadata_v2())),
                 ForkName::Base,
             ),
```

Comment:
Would be good to test the blob protocols here as well. Can do it in a separate PR too later though

Reply:
Interestingly the tests below failed with error `Err(InvalidData("RPC response length is out of bounds, length 131256, max 1125899911330948, min 131660"))`. Will take a look later this evening.

```rust

        assert_eq!(
            encode_then_decode_response(
                SupportedProtocol::BlobsByRangeV1,
                RPCCodedResponse::Success(RPCResponse::BlobsByRange(default_blob_sidecar())),
                ForkName::Deneb,
            ),
            Ok(Some(RPCResponse::BlobsByRange(default_blob_sidecar()))),
        );

        assert_eq!(
            encode_then_decode_response(
                SupportedProtocol::BlobsByRootV1,
                RPCCodedResponse::Success(RPCResponse::SidecarByRoot(default_blob_sidecar())),
                ForkName::Deneb,
            ),
            Ok(Some(RPCResponse::SidecarByRoot(default_blob_sidecar()))),
        );
    }
```

Reply:
So the first test above passes, but 2nd one (`BlobsByRoot`) doesn't - I suspect there might be an issue somewhere in the merge 👀  might be worth getting it fixed before merging this PR.

Reply:
ah ok looks like it's failing because the ssz length bounds for `BlobsByRoot` RPC messages seems to be wrong hence giving an out of bound error

https://github.com/sigp/lighthouse/blob/03a17a84dac0df9d0d76e0689c4f03b3643414fc/beacon_node/lighthouse_network/src/rpc/protocol.rs#L426

Reply:
Test and fix added in https://github.com/sigp/lighthouse/pull/4432/commits/03a17a84dac0df9d0d76e0689c4f03b3643414fc.

Reply:
Thanks jimmy! Sorry I think that was fixed in the devnet 6 branch, but hadn't made it here yet